golang/x/relay/keeper/querier.go

```diff
@@ -34,48 +73,242 @@ func queryIsAncestor(ctx sdk.Context, path []string, req abci.RequestQuery, keep
 
 	// getAncestor is removed by the handler switch/case block above
 
-	// parse the first path item as hex.
-	// TODO: check that the path is this many items long, error if not
-	hexDigest := path[0]
-	digestBytes, decodeErr := hex.DecodeString(hexDigest)
-	if decodeErr != nil {
-		return []byte{}, types.ErrBadHex(types.DefaultCodespace)
+	// check that the path is this many items long, error if not
+	if len(path) > 3 {
+		return []byte{}, types.ErrTooManyArguments(types.DefaultCodespace)
+	} else if len(path) < 2 {
+		return []byte{}, types.ErrNotEnoughArguments(types.DefaultCodespace)
 	}
-	digestLE, newDigestErr := btcspv.NewHash256Digest(digestBytes)
-	if newDigestErr != nil {
-		return []byte{}, types.FromBTCSPVError(types.DefaultCodespace, newDigestErr)
+
+	// parse the first path item as hex.
+	digestLE, digestErr := hash256DigestFromHex(path[0])
+	if digestErr != nil {
+		return []byte{}, digestErr
 	}
 
 	// parse the second path item as hex.
-	// TODO: check that the path is this many items long, error if not
-	ancestorDigest := path[1]
-	ancestorDigestBytes, decodeErr := hex.DecodeString(ancestorDigest)
-	if decodeErr != nil {
-		return []byte{}, types.ErrBadHex(types.DefaultCodespace)
-	}
-	ancestorDigestLE, newDigestErr := btcspv.NewHash256Digest(ancestorDigestBytes)
-	if newDigestErr != nil {
-		return []byte{}, types.FromBTCSPVError(types.DefaultCodespace, newDigestErr)
+	ancestor, ancestorErr := hash256DigestFromHex(path[1])
+	if ancestorErr != nil {
+		return []byte{}, ancestorErr
 	}
 
-	// TODO: check that the path is this many items long, error if not
-	limit := uint32(25) // TODO: parse from path, use 240 as default if not in path
+	var limit uint32
```

Comment:
good spot for a convenience function. should take

```
func DecodeUint32FromPath(path []string, idx int, default uint32) (uint32, sdk.Error) {
   if len(path) >= idx {
        return default, nil
   }
   // parse int from path[idx], return error if necessary
}
```

Reply:
ok, added